concordium-consensus/src/Concordium/Scheduler/ProtocolLevelTokens/Module.hs

```diff
@@ -90,254 +94,249 @@ initializeToken tokenParam = do
+                            success <- transfer tcSender recipientAccount pthoAmount pthoMemo
+                            unless success $ do
                                 availableBalance <- getAccountBalance tcSender
                                 failTH
                                     TokenBalanceInsufficient
                                         { trrOperationIndex = opIndex,
                                           trrAvailableBalance = toTokenAmount decimals availableBalance,
-                                          trrRequiredBalance = ptgoUnprocessedAmount
+                                          trrRequiredBalance = ttAmount pthoUnprocessed
                                         }
-                        PTGOTokenAddAllowList{..} -> do
-                            requireFeature opIndex "addAllowList" "allowList"
-                            account <- requireAccount opIndex ptgoTarget
-                            pltTickEnergy tokenListOperationCost
-                            setAccountState account "allowList" (Just "")
-                            logEncodeTokenEvent (AddAllowListEvent ptgoTarget)
-                        PTGOTokenRemoveAllowList{..} -> do
-                            requireFeature opIndex "removeAllowList" "allowList"
-                            account <- requireAccount opIndex ptgoTarget
-                            pltTickEnergy tokenListOperationCost
-                            setAccountState account "allowList" Nothing
-                            logEncodeTokenEvent (RemoveAllowListEvent ptgoTarget)
-                        PTGOTokenAddDenyList{..} -> do
-                            requireFeature opIndex "addDenyList" "denyList"
-                            account <- requireAccount opIndex ptgoTarget
-                            pltTickEnergy tokenListOperationCost
-                            setAccountState account "denyList" (Just "")
-                            logEncodeTokenEvent (AddDenyListEvent ptgoTarget)
-                        PTGOTokenRemoveDenyList{..} -> do
-                            requireFeature opIndex "removeDenyList" "denyList"
-                            account <- requireAccount opIndex ptgoTarget
-                            pltTickEnergy tokenListOperationCost
-                            setAccountState account "denyList" Nothing
-                            logEncodeTokenEvent (RemoveDenyListEvent ptgoTarget)
+                        tokenGovernanceOp -> do
+                            mbGovAccountIx <- getTokenState "governanceAccount"
+                            case mbGovAccountIx of
+                                Nothing -> error "Invariant violation: Token state is missing the governance account address."
+                                Just govAccountIx -> do
+                                    senderIx <- getAccountIndex tcSender
+                                    unless (govAccountIx == encode senderIx) $
```

Comment:
I think it's bad practice to compare encodings rather than the model values. Instead:
```suggestion
                                    unless (decode govAccountIx == Right senderIx) $
```